### PR TITLE
feat: universal dialog component

### PR DESCRIPTION
**Pull request by creed-victor**
https://sovryn.monday.com/boards/2218344956/pulses/2236624785

Example is in https://feat-refactor-dialog.sovryn.pages.dev/storybook?path=/story/molecule-dialog

Note: had to make close button / icon smaller than it was, otherwise it's going out of viewport on smaller screens (because it's outside of dialog for non-mobile devices)

### CODE REVIEW

**Review by JoeStorm99 — DISMISSED** [review]

File: src/app/components/Dialog/index.module.scss [hunk 1]
```diff
@@ -0,0 +1,31 @@
+.dialog_wrapper {
+    @apply tw-fixed tw-inset-0;
+    @apply tw-flex tw-flex-col tw-justify-center tw-items-center;
```
Comment (tiltom) [inline src/app/components/Dialog/index.module.scss:3]:
```suggestion
    @apply tw-fixed tw-inset-0 tw-flex tw-flex-col tw-justify-center tw-items-center;
```

File: src/app/components/Dialog/index.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  useMemo,
+  MouseEvent,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+import classNames from 'classnames';
+import { translations } from 'locales/i18n';
+import { Overlay, OverlayProps } from '../Overlay';
+import styles from './index.module.scss';
+
+interface DialogFunctionComponent<T = {}> extends FunctionComponent<T> {
```
Comment (tiltom) [inline src/app/components/Dialog/index.tsx:14]:
```suggestion
interface IDialogFunctionComponent<T = {}> extends FunctionComponent<T> {
```

Comment (tiltom) [inline src/app/components/Dialog/index.tsx:14]: Create `types.ts` file and extract this, `DialogSize` and `sizeMap` to it.

File: src/app/components/Dialog/index.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  onClose,
+}) => {
+  const { t } = useTranslation();
+  const sizeClassNames = useMemo(() => sizeMap[width], [width]);
+
+  const handleClose = useCallback(() => {
+    if (onClose) {
+      onClose();
+    }
+  }, [onClose]);
```
Comment (tiltom) [inline src/app/components/Dialog/index.tsx:66]:
```suggestion
const handleClose = useCallback(() => {
    onClose?.();
  }, [onClose]);
```

File: src/app/components/Dialog/index.module.scss [hunk 4]
```diff
@@ -0,0 +1,31 @@
+.dialog_wrapper {
```
Comment (tiltom) [inline src/app/components/Dialog/index.module.scss:1]:
```suggestion
.wrapper {
```

File: src/app/components/Dialog/index.module.scss [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    @apply tw-fixed tw-inset-0;
+    @apply tw-flex tw-flex-col tw-justify-center tw-items-center;
+}
+
+.dialog_container {
```
Comment (tiltom) [inline src/app/components/Dialog/index.module.scss:6]:
```suggestion
container {
```

File: src/app/components/Dialog/index.module.scss [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+.dialog {
+    @apply tw-relative tw-bg-black tw-text-sov-white tw-px-4 tw-py-8 tw-mx-auto tw-min-h-full;
+    @screen sm {
+       @apply tw-min-h-0 tw-rounded-xl tw-my-16 tw-py-8;
+    }
+}
+
+.close_button {
```
Comment (tiltom) [inline src/app/components/Dialog/index.module.scss:20]:
```suggestion
.closeButton {
```

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by tiltom — DISMISSED** [review]

File: src/app/components/Dialog/types.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import type { FunctionComponent } from 'react';
+
+export interface IDialogFunctionComponent<T = {}> extends FunctionComponent<T> {
+  index: number;
+}
```
[the remark this answers is not in the document]
Comment (creed-victor) [reply src/app/components/Dialog/types.ts:5]: Not sure I understood. It's static variable which tracks how many dialogs was opened, so if you would open another dialog while one is already opened - newest would appear on top of the previous one

**Review by JoeStorm99 — DISMISSED** [review]

**Review by soulBit — CHANGES_REQUESTED** [review]
Generally looks good but few things to check before I approve:

1. Point 6a in the spec indicates we should have sizes for `2-7xl` but we only have up to 3 in this PR
2. Point 6c in the spec doesn't seem to be implemented here

@ShynRou if the above two points are fine then let's update the spec to match

I've also made a small fix to the `DialogSize` enum so that the size values appear correctly in Storybook (they were appearing as indexes previously).

**Comment by creed-victor** [comment]
> 1. Point 6a in the spec indicates we should have sizes for `2-7xl` but we only have up to 3 in this PR

Added missing sizes

> 2. Point 6c in the spec doesn't seem to be implemented here

It's implemented but you need to change your storybook preview width to the mobile device (xs size)

**Review by tiltom — APPROVED** [review]

**Review by soulBit — APPROVED** [review]

**Review by JoeStorm99 — APPROVED** [review]

**Pull request merged**